qt/widgets/common/src/FitScriptGeneratorModel.cpp

```diff
@@ -0,0 +1,292 @@
+#include "MantidQtWidgets/Common/FitScriptGeneratorModel.h"
+
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/CompositeFunction.h"
+#include "MantidAPI/FunctionFactory.h"
+#include "MantidAPI/IFunction.h"
+#include "MantidAPI/MatrixWorkspace.h"
+
+#include <boost/algorithm/string.hpp>
+
+#include <algorithm>
+#include <iterator>
+#include <stdexcept>
+
+using namespace Mantid::API;
+
+namespace {
+
+IFunction_sptr createIFunction(std::string const &functionString) {
+  return FunctionFactory::Instance().createInitialized(functionString);
+}
+
+CompositeFunction_sptr toComposite(IFunction_sptr function) {
+  return std::dynamic_pointer_cast<CompositeFunction>(function);
+}
+
+CompositeFunction_sptr createEmptyComposite() {
+  return toComposite(createIFunction("name=CompositeFunction"));
+}
+
+MultiDomainFunction_sptr
+createMultiDomainFunction(std::size_t const &numberOfDomains) {
+  return FunctionFactory::Instance().createInitializedMultiDomainFunction(
+      "name=CompositeFunction", numberOfDomains);
+}
+
+std::vector<std::string> splitStringBy(std::string const &str,
+                                       std::string const &delimiter) {
+  std::vector<std::string> subStrings;
+  boost::split(subStrings, str, boost::is_any_of(delimiter));
+  subStrings.erase(std::remove_if(subStrings.begin(), subStrings.end(),
+                                  [](std::string const &subString) {
+                                    return subString.empty();
+                                  }),
+                   subStrings.cend());
+  return subStrings;
+}
+
+bool isSinglePrefix(std::string const &prefix) {
+  auto const subStrings = splitStringBy(prefix, "f.");
+  return subStrings.size() == 1;
+}
+
+std::size_t getPrefixIndexAt(std::string const &prefix,
+                             std::size_t const &index) {
+  auto const subStrings = splitStringBy(prefix, "f.");
+  return std::stoull(subStrings[index]);
+}
+
+std::string getTopPrefix(std::string const &prefix) {
+  auto topPrefix = prefix;
+  auto const firstDotIndex = topPrefix.find(".");
+  if (firstDotIndex != std::string::npos)
+    topPrefix.erase(firstDotIndex, topPrefix.size() - firstDotIndex);
+  return topPrefix;
+}
+
+IFunction_sptr getFunctionAtPrefix(std::string const &prefix,
+                                   IFunction_sptr const &function,
+                                   bool isLastFunction = false) {
+  if (isLastFunction)
+    return function;
+
+  auto const isLast = isSinglePrefix(prefix);
+  auto const topPrefix = getTopPrefix(prefix);
+  auto const firstIndex = getPrefixIndexAt(prefix, 0);
+
+  try {
+    return getFunctionAtPrefix(topPrefix, function->getFunction(firstIndex),
+                               isLast);
+  } catch (std::exception const &) {
+    return IFunction_sptr();
+  }
+}
+
+} // namespace
+
+namespace MantidQt {
+namespace MantidWidgets {
+
+/**
+ * FitDomain struct methods.
+ */
+
+FitDomain::FitDomain(std::string const &prefix,
```

Comment:
is it worth putting these into a seperate file so they dont get lost at the end of this one?

Reply:
This has changed in the next PR quite a lot. It probably wouldn't be worth moving it (at least not for now - maybe in a later PR if it gets more complicated)

Reply:
ok